www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+```diff
+/* Add JavaScript code here! */
+- console.log('Hello World! You did! Welcome to Snowpack :D');
++import React from 'react';
++import ReactDOM from 'react-dom';
++ReactDOM.render(<div>"HELLO REACT"</div>, document.getElementById('root'));
+```
+
+If you run this code you'll get `Error: Target container is not a DOM element.` The call to `document.getElementById('root')` is failing because the HTML page has no element with that ID. Head to index.html and add one in the `<body>`
+
+```diff
+  <body>
+-    <h1>Welcome to Snowpack!</h1>
++    <div id="root"></div>
+    <script type="module" src="/index.js"></script>
+```
+
+Now it should run and you should see "HELLO REACT"
+
+Image: screenshot of the project, which shows "HELLO REACT" on a white background
+
+## Organizing your files
+
+If you're going to be adding a bunch of new files, you probably don't want them in the root directory. A standard way of organizing React projects is an `src` directory for components and their associated files. Then a public directory for static assets like images, fonts, icons, and sometimes CSS. To do this you'll need to reorganize your files, then tell Snowpack about it in the configuration file, and update your paths in index.html.
+
+```
+📁 src
+↳ index.jsx
+📁 public
+↳ index.css
+↳ index.html
+```
+
+```bash
+mkdir src
+mkdir public
+mv index.jsx src/index.jsx
+mv index.html public/index.html
+mv index.css public/index.css
+```
+
+You need to tell Snowpack about this new structure using the [Snowpack Configuration API](https://www.snowpack.dev/#configuration) in the `snowpack.config.js` file. Right now it's a bare bones scaffold that doesn't do anything. Add this to the `mount` object:
+
+```diff
+  mount: {
+-    /* ... */
++    public: '/',
++    src: '/_dist_',
+  },
+```
+
+This tells it look at `public` instead of look at `/` for the index. For example when building `index.css`, Snowpack looks for this files in `./public/index.css`.
+
+And it also tells it to put the final output files in a subdirectory called `_dist_` so change that path in `index.html` from `/index.js` to `/_dist_/index.js`.
+
+```diff
+    <h1>Welcome to Snowpack!</h1>
+    <div id="root"></div>
+-    <script type="module" src="/index.js"></>
++    <script type="module" src="/_dist_/index.js"></script>
+  </body>
+```
+
+If Snowpack is still running, exit the process, you need to restart Snowpack for configuration file changes. When you start up again, if it worked, it should look the same.
+
+Create a new file at `src/App.jsx` and paste the following code into this new file to create an `App` component:
+
+```jsx
+import React, {useState, useEffect} from 'react';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
```

Comment:
someone reading this probably already knows react. since we're in a guide, should try to keep the amount of code minimal.

Reply:
A stated goal here was to get you something that matched the current `create-snowpack-app` React template. If you see that template, you'll see it's basically an exact match to what you end up with here.

We also need the counter to highlight HMR & Fast Reload in the section below/

I'm personally not against the extra lines of copy-paste here, but @melissamcewen it could be worth revisiting/confirming this goal. Do we need to match the create-snowpack-app react template EXACTLY, or can we just match it in spirit? If just in spriit, then I'd agree we could probably trim some CSS, or include a Snowpack logo in the `minimal` template and then use that here instead of the React logo.